supervisor/api/utils.py

```diff
@@ -129,12 +129,15 @@ def api_return_error(
         JSON_RESULT: RESULT_ERROR,
         JSON_MESSAGE: message or "Unknown error, see supervisor",
     }
-    if isinstance(error, APIError) and error.job_id:
-        result[JSON_JOB_ID] = error.job_id
+    status = 400
+    if isinstance(error, APIError):
+        status = error.status
+        if error.job_id:
+            result[JSON_JOB_ID] = error.job_id
```

Comment:
This is technically unrelated to the current PR, just something I discovered while writing the test cases. But it felt really wrong to me that the `APIForbidden` exception did not actually return `403 Forbidden` and returned `400 Bad Request`. Especially since we seemed to use it as a way to convey to clients that they did something not allowed and that wasn't being translated to the status code.

Although this does change the status code in a few places, I don't think this is really a backwards compatibility issue. It's still a 400 range error indicating the client did something wrong and its a status code we do use they should already be looking for.

Reply:
I fully agree with this finding and the change. However, I'd prefer if this is a separate PR, even though it is very small. Mostly about having a proper entry for this change in the release changelog. Also in case we have to revert (even though rather unlikely).

Reply:
@agners ok, created https://github.com/home-assistant/supervisor/pull/4943